stdlib/public/SDK/Foundation/NSArray.swift

```diff
@@ -106,6 +106,14 @@ extension Array : _ObjectiveCBridgeable {
   }
 }
 
+extension NSArray : _HasCustomAnyHashableRepresentation {
```

Comment:
@parkera I just saw this, does it need bringing to SCLF?

Reply:
Yes, I think we will need to do that.

Reply:
Sorry I missed this! Here is a list of `_HasCustomAnyHashableRepresentation` conformances in the Foundation overlay that are currently missing in open-source Foundation:

```
NSAffineTransform
NSArray
NSCalendar
NSCharacterSet
NSDate
NSDateComponents
NSDateInterval
NSDictionary
NSLocale
NSNotification
NSNumber
NSSet
NSString
NSTimeZone
NSURL
NSURLComponents
NSURLQueryItem
NSURLRequest
```

This PR added NSArray and NSDictionary to the list. The intention behind _HCAHR conformances is to have the AnyHashable representation of these classes hash the same way as the corresponding Swift value type. (I don't know if this is as much of a concern in swift-corelibs-foundation as it is in Cocoa, though.)

Reply:
OK that's quite a long list. @parkera how do you suggest we proceed with resolving this?

Reply:
I think it's only really critical if/when bridging is enabled on Linux; otherwise, it's okay for, say, NSString and String to hash differently when put into an AnyHashable.

Reply:
We really need to strive to keep things in sync. @lorentey is it possible to apply your PR to the swift-corelibs APIs?

Reply:
@parkera It looks like most of these implementations would easily translate over to corelibs. `as`-conversions won't work without bridging, but it seems straightforward to replace them with explicit conversions. 

I expect most of the effort would be spent on writing new tests to cover the new functionality.

Reply:
Alright. Part of the deal of contributing to the Foundation part of the overlay is that we also need those changes for Linux. We can't let it lag behind.

Reply:
Agreed; I created a bug to track this effort: https://bugs.swift.org/browse/SR-7436

Reply:
Thanks @lorentey - are you going to resolve this bug?

Reply:
@ianpartridge It turns out conforming to _HCAHR currently has no effect on platforms without the Objective-C runtime. I'll look into whether this can be lifted (as part of SR-7436). In the meantime, I don't think it's worth adding these conformances -- it'd be dead code.